### ISSUE TITLE
Unexpected errors on REST JavaScript/TypeScript projects

### ISSUE THREAD

**Issue by jose** [EMResearch/EMB#41]
Hi @arcuri82,

I've been getting some unexpected errors on some REST JavaScript/TypeScript projects (`js-rest-scs`, `disease-sh-api`, and `realworld-app`).  Please find below the exceptions/erros I've been getting.

### `realworld-app`

```
start db with test container
connecting mysql://localhost:29464/test
Started EM Controller on port 10000
^[[32m[Nest] 579854   - ^[[39m04/13/2022, 1:23:03 PM   ^[[38;5;3m[NestFactory] ^[[39m^[[32mStarting Nest application...^[[39m
^[[32m[Nest] 579854   - ^[[39m04/13/2022, 1:23:03 PM   ^[[38;5;3m[InstanceLoader] ^[[39m^[[32mTypeOrmModule dependencies initialized^[[39m^[[38;5;3m +57ms^[[39m
^[[31m[Nest] 579854   - ^[[39m04/13/2022, 1:23:03 PM   ^[[38;5;3m[TypeOrmModule] ^[[39m^[[31mUnable to connect to the database. Retrying (1)...^[[39m^[[38;5;3m +17ms^[[39m
/home/jose/tests-data/tools/emb/dist/realworld-app/src/src/article/article.entity.ts:1
import { Entity, PrimaryGeneratedColumn, Column, OneToOne, ManyToOne, OneToMany, JoinColumn, AfterUpdate, BeforeUpdate } from 'typeorm';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1032:15)
    at Module._compile (node:internal/modules/cjs/loader:1067:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at /home/jose/tests-data/tools/emb/dist/realworld-app/node_modules/typeorm/util/DirectoryExportedClassesLoader.js:41:39
    at Array.map (<anonymous>)
```

### `disease-sh-api`

```
start docker db
/home/jose/tests-data/tools/emb/dist/disease-sh-api/node_modules/testcontainers/dist/docker-compose/functions/docker-compose-up.js:36
        throw new Error(err);
              ^

Error: Error: spawn docker-compose ENOENT
    at /home/jose/tests-data/tools/emb/dist/disease-sh-api/node_modules/testcontainers/dist/docker-compose/functions/docker-compose-up.js:36:15
    at Generator.throw (<anonymous>)
    at rejected (/home/jose/tests-data/tools/emb/dist/disease-sh-api/node_modules/testcontainers/dist/docker-compose/functions/docker-compose-up.js:6:65)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### `js-rest-scs`

```
Started EM Controller on port 10040
ReferenceError: parseDouble is not defined
    at /home/jose/tests-data/tools/emb/dist/js-rest-scs/src/app.js:78:18
    at Layer.handle [as handle_request] (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/layer.js:95:5)
    at /home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:281:22
    at param (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:354:14)
    at param (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:365:14)
    at param (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:365:14)
    at param (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:365:14)
ReferenceError: rep is not defined
    at /home/jose/tests-data/tools/emb/dist/js-rest-scs/src/app.js:106:17
    at Layer.handle [as handle_request] (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/layer.js:95:5)
    at /home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:281:22
    at param (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:354:14)
    at param (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:410:3)
    at next (/home/jose/tests-data/tools/emb/dist/js-rest-scs/node_modules/express/lib/router/index.js:275:10)
```

```
$ node --version
v16.14.2

$ npm --version
8.5.0

$ java -version
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)
```

Could those be related to the Node.js / NPM version I'm currently using?  If so, which one would you recommend?

--
Best,
Jose

**Comment by arcuri82** [comment]
Hi,
  which version of EMB are u using? `1.4.0` or `1.4.1-SNAPSHOT` in `develop` branch?
Regarding JS support, we have a recent  paper at [ICST](https://github.com/EMResearch/EvoMaster/blob/master/docs/publications/2022_icst.pdf), but we are not done yet in writing the documentation for EMB...  (eg how to build the TS projects and run EM drivers) 
`npm` versions shouldn't matter. for `node` we use both `14` and `16`

**Comment by jose** [comment]
Completly forgot to provide that information, sorry.  I'm using the latest version in the `master` branch (v1.4.0, [de2cab3](https://github.com/EMResearch/EMB/tree/de2cab3e4431672cad9c49ad9ab040505d271239)).

Regarding instructions on how to build the TS projects, I've been following the instructions/commands in the [scripts/dist.py](https://github.com/EMResearch/EMB/blob/master/scripts/dist.py#L161) script, and on how to run the EM drivers I've been following the instructions/commands in the [scripts/exp.py](https://github.com/EMResearch/EvoMaster/blob/master/scripts/exp.py#L442) script.

**Comment by arcuri82** [comment]
hmmm... looks like when I added those JS SUTs to EMB I didn't fully tested them. i m in the process of refactoring/checking all these JS SUTs, but that it is currently under `develop` branch (but it is not completed). i guess until we make a new `1.5.0` release (in some weeks), those JS SUTs should not be used (unless you want to check the `develop` branch, but that is work in progress, or look at the replication package of ICST)

**Comment by jose** [comment]
Ok, thanks.  I will, for now, use the Java projects and leave the evaluation on the JS projects for later.

**Comment by arcuri82** [comment]
this issue should had been solved in the new 1.5.0 release
